plugin/src/main/kotlin/org/gradle/github/dependency/GitHubDependencySubmissionPlugin.kt

```diff
@@ -0,0 +1,28 @@
+package org.gradle.github.dependency
+
+import org.gradle.api.Plugin
+import org.gradle.api.internal.GradleInternal
+import org.gradle.api.invocation.Gradle
+import org.gradle.api.logging.Logging
+import org.gradle.github.dependency.extractor.ForceDependencyResolutionPlugin
+import org.gradle.github.dependency.extractor.GithubDependencyExtractorPlugin
+
+@Suppress("unused")
+class GitHubDependencySubmissionPlugin : Plugin<Gradle> {
+
+    private val LOGGER = Logging.getLogger(GitHubDependencySubmissionPlugin::class.java)
+
+    override fun apply(gradle: Gradle) {
+        LOGGER.lifecycle("Applying Plugin: GitHubDependencySubmissionPlugin")
+        val buildPath = (gradle as GradleInternal).identityPath
+        // Only apply the dependency extractor to the root build
+        if (gradle.parent == null) {
+            LOGGER.info("Applying Plugin: GitHubDependencyExtractorPlugin to root build")
+            gradle.pluginManager.apply(GithubDependencyExtractorPlugin::class.java)
+        }
+
+        // Apply the dependency resolver to all builds
+        LOGGER.info("Applying Plugin: ForceDependencyResolutionPlugin to build $buildPath")
+        gradle.pluginManager.apply(ForceDependencyResolutionPlugin::class.java)
```

Comment:
Doesn't this prevent you from testing variants where all dependencies aren't resolved? For example, some of the existing tests that have been removed.

Reply:
I don't understand what you mean by "variants where all dependencies aren't resolved".

Reply:
The end goal for this work is #20. The `ForceDependencyResolutionPlugin` achieves this by adding a new, default task to the build that will do as much resolution as possible without requiring the project to be actually compiled and built.

I think you mean that we should continue to test that all resolved dependencies are collected when a different task is executed, one which may only resolve a subset of the build dependencies. I'll be sure to retain some coverage for this case.